opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/ResultHandler.java

```diff
@@ -0,0 +1,51 @@
+package de.adorsys.opba.protocol.facade.services;
+
+import de.adorsys.opba.db.domain.entity.sessions.AuthSession;
+import de.adorsys.opba.db.domain.entity.sessions.ServiceSession;
+import de.adorsys.opba.db.repository.jpa.AuthenticationSessionRepository;
+import de.adorsys.opba.protocol.api.dto.result.RedirectionResult;
+import de.adorsys.opba.protocol.api.dto.result.Result;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityManager;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class ResultHandler {
+
+    private final EntityManager entityManager;
+    private final AuthenticationSessionRepository authenticationSessions;
+
+    @Transactional
+    public <R extends Result<?>> R handleResult(R result, UUID serviceSessionId) {
+        if (!(result instanceof RedirectionResult)) {
+            return result;
+        }
+
+        return handleRedirect(result, serviceSessionId);
+    }
+
+    private <R extends Result<?>> R handleRedirect(R result, UUID serviceSessionId) {
+        // Auth session is 1-1 to service session, using id as foreign key
```

Comment:
This assumption may be wrong. User generally can have multiple authorisation sessions.

To keep it simple it's OK leaving it as is for now, maybe it's worth to add FIXME or TODO and create a ticket for this use case for future.

Reply:
User can have many authorization sessions, that's true, but why one ServiceSession can have more that one (active) Authorization sessions

Reply:
It will strongly depend on many parameters like: consent lifetime on the bank side, workflow on Fintech side, implementation details on Fintech side.

It can be, that this is a consent prolongation, not the first consent initiation. For the Fintech this is still the same ServiceSession.

It can also be, that user accidentally failed the authorisation, by entering false TAN for example.
Or Authorisation was broken due to network connection problems.

If Fintech is dump (and many will be), then reloading user page in browser may cause repeating of service request from Fintech to us inside one session.

Reply:
1. It can be, that this is a consent prolongation, not the first consent initiation. For the Fintech this is still the same ServiceSession.
- In this case, it is a new (active) authorization session - old one must be removed and discarded as it is no longer valid (why to keep it inside active list it is finished -> delete or move to history list).

1. It can also be, that user accidentally failed the authorisation, by entering false TAN for example. Or Authorisation was broken due to network connection problems.
- False TAN:  It is the continuation of an already existing Authorization session that is handled by protocol, from our side we can't simply replay user input, in this case, we can provide protocol context persistence for continuation
- Network problems - a continuation of already existing authorization session (we have its context in DB, we can do it)

1. If Fintech is dump (and many will be), then reloading user page in browser may cause repeating of service request from Fintech to us inside one session.
- In this case, if consent was already granted - there will be no authorization session as the protocol will find consent. If not yet granted - we see that there is an open authorization session and simply continue it

Reply:
Then we are talking here not about the ServiceSession (identified by Service-Session-Id), but about authorization session. Let's discuss it with Francis to get good terminology